src/tuwien/auto/calimero/knxnetip/util/Srp.java

```diff
@@ -0,0 +1,288 @@
+
+    Linking this library statically or dynamically with other modules is
+    making a combined work based on this library. Thus, the terms and
+    conditions of the GNU General Public License cover the whole
+    combination.
+
+    As a special exception, the copyright holders of this library give you
+    permission to link this library with independent modules to produce an
+    executable, regardless of the license terms of these independent
+    modules, and to copy and distribute the resulting executable under terms
+    of your choice, provided that you also meet, for each linked independent
+    module, the terms and conditions of the license of that module. An
+    independent module is a module which is not derived from or based on
+    this library. If you modify this library, you may extend this exception
+    to your version of the library, but you are not obligated to do so. If
+    you do not wish to do so, delete this exception statement from your
+    version.
+*/
+
+package tuwien.auto.calimero.knxnetip.util;
+
+import tuwien.auto.calimero.KNXFormatException;
+import tuwien.auto.calimero.KNXIllegalArgumentException;
+
+import java.nio.ByteBuffer;
+
+/**
+ * Search Request Parameter Block (SRP).
+ * <p>
+ * A SRP is used to transfer additional information in a KNXnet/IP extended search request.
+ * <p>
+ * The mandatory flag of the SRP can be used to increase or limit the responses received
+ * from an extended search request. If the mandatory flag of the SRP is set, KNXnet/IP router
+ * or server will only respond to the extended search request if the search request parameter
+ * block is completely satisfied.
+ * <p>
+ *
+ * @author Karsten Heimrich
+ */
+public final class Srp
+{
+	public enum Type {
+		/**
+		 * The invalid type is used to test the behavior of the KNXnet/IP router or server for unknown
+		 * SRPs. Please do not use in production type.
+		 */
+		InvalidSrp(0x00),
+
+		/**
+		 * The select by programming mode type-type is used to indicate that a client is interested only in the
+		 * response from KNXnet/IP router or server where programming mode is currently enabled.
+		 */
+		SelectByProgrammingMode(0x01),
+
+		/**
+		 * The select by MAC address type-type is used to indicate that a client is interested only in the
+		 * response from KNXnet/IP router or server with the given MAC address.
+		 */
+		SelectByMacAddress(0x02),
+
+		/**
+		 * The select by service type-type is used to indicate that a client is interested only in the response
+		 * from KNXnet/IP router or server supporting the given service family and in at least the given version.
+		 */
+		SelectByService(0x03),
+
+		/**
+		 * The request DIBs type-type is used to indicate that a client is interested in at least the given
+		 * description types.
+		 */
+		RequestDibs(0x04);
+
+		private final int type;
+
+		public final int getValue() {
+			return type;
+		}
+
+		public static Type from(final int code) {
```

Comment:
ditto, AFAIKS this can remain an implementation detail

Reply:
Fixed.